File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let start = 0;
    let next = 1;
    let duplicates = 0;
    let end = nums.length - 1;
    while (next <= end) {
        if (nums[start] != nums[next]) {
            let temp = nums[start+1];
            nums[start+1] = nums[next];
            nums[next]= temp;
            start++;
        } else {
            duplicates++;
        }
        next++;
    }
    return nums.length - duplicates;
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        start = 0
        next = 1
        duplicates = 0
        end = len(nums) - 1
        while next <= end:
            if nums[start] != nums[next]:
                temp = nums[start+1]
                nums[start+1] = nums[next]
                nums[next] = temp
                start += 1
            else:
                duplicates += 1
            next += 1
        return len(nums) - duplicates